File: reference/go/aep/fixtures.go
package aep

import (
	"encoding/json"
	"os"
	"strings"
)

type ManifestFixture struct {
	Path          string   `json:"path"`
	Level         string   `json:"level"`
	Description   string   `json:"description"`
	Expectation   string   `json:"expectation"`
	Tags          []string `json:"tags"`
	ExpectedTypes []string `json:"expected_types"`
}

type Manifest struct {
	AEPVersion         string           `json:"aep_version"`
	DefaultTargetLevel string           `json:"default_target_level"`
	Levels             []string         `json:"levels"`
	Fixtures           []ManifestFixture `json:"fixtures"`
}

func LoadManifest(path string) (*Manifest, error) {
	data, err := os.ReadFile(path)
	if err != nil {
		return nil, err
	}
	var manifest Manifest
	if err := json.Unmarshal(data, &manifest); err != nil {
		return nil, err
	}
	return &manifest, nil
}

func LoadFixture(path string) ([]map[string]any, error) {
	data, err := os.ReadFile(path)
	if err != nil {
		return nil, err
	}
	lines := strings.Split(strings.TrimSpace(string(data)), "\n")
	var events []map[string]any
	for _, line := range lines {
		if line == "" {
			continue
		}
		var event map[string]any
		if err := json.Unmarshal([]byte(line), &event); err != nil {
			return nil, err
		}
		events = append(events, event)
	}
	return events, nil
}

File: reference/java/src/main/java/com/axisrobo/aep/Fixtures.java
package com.axisrobo.aep;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.core.type.TypeReference;
import com.fasterxml.jackson.databind.ObjectMapper;
import java.io.IOException;
import java.nio.file.Files;
import java.nio.file.Path;
import java.util.*;
import java.util.stream.Collectors;

public final class Fixtures {
    private static final ObjectMapper MAPPER = new ObjectMapper();

    @JsonIgnoreProperties(ignoreUnknown = true)
    public record ManifestFixture(String path, String level, String description,
                                  String expectation, List<String> tags,
                                  @JsonProperty("expected_types") List<String> expectedTypes,
                                  @JsonProperty("expected_stats") Map<String, Object> expectedStats) {}

    public record Manifest(@JsonProperty("aep_version") String aep_version,
                           @JsonProperty("default_target_level") String default_target_level,
                           List<String> levels, List<ManifestFixture> fixtures) {}

    public static Manifest loadManifest(String path) throws IOException {
        return MAPPER.readValue(Path.of(path).toFile(), Manifest.class);
    }

    public static List<Map<String, Object>> loadFixture(String path) throws IOException {
        var text = Files.readString(Path.of(path));
        return Arrays.stream(text.strip().split("\n"))
            .filter(line -> !line.isBlank())
            .map(line -> {
                try {
                    return MAPPER.<Map<String, Object>>readValue(line, new TypeReference<>() {});
                } catch (IOException e) {
                    throw new RuntimeException("invalid NDJSON line", e);
                }
            })
            .collect(Collectors.toList());
    }
}
